### PR TITLE
Support multiple edX installations

### PR DESCRIPTION
**Pull request by blarghmatey**
As part of our work to upgrade our xPro and residential MITx environments to the Maple build we need a way to support multiple different (and sometimes shared) releases of Open edX across different environments and with different configurations. This adds support for being able to build the requisite AMIs that we can use to deploy to the different target environments.

### CODE REVIEW

File: src/bilder/images/edxapp/templates/edxapp/xpro/studio_only.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# ALTERNATE_WORKER_QUEUES: lms  # Already has a sane default in the code
+ADDL_INSTALLED_APPS:  # ADDED KEY
+  - git_auto_export
+COURSE_AUTHORING_MICROFRONTEND_URL:
+COURSE_IMPORT_EXPORT_BUCKET: {{ key "edxapp/s3-course-bucket" }}  # MODIFIED
+GIT_REPO_EXPORT_DIR: /edx/var/edxapp/export_course_repos
+GIT_EXPORT_DEFAULT_IDENT:
+  name: MITx Online
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/xpro/studio_only.yml:8]: This is xPRO, no?

File: src/bilder/images/edxapp/templates/edxapp/xpro/studio_only.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  email: mitx-devops@mit.edu
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/xpro/studio_only.yml:9]: Shouldn't we be using `odl-devops` instead?

File: src/bilder/images/edxapp/packer_vars/mitx.pkrvars.hcl [hunk 3]
```diff
@@ -0,0 +1,3 @@
+build_environment    = "mitx-ci"
```
Comment (shaidar) [inline src/bilder/images/edxapp/packer_vars/mitx.pkrvars.hcl:1]: Does the filename need to include the `ci` in it?

File: src/bilder/images/edxapp/group_data/all.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        },
+    }
+]
+
+edx_plugins = {
+    "mitxonline": [
+        "celery-redbeat",  # Support for using Redis as the lock for Celery schedules
+        "django-redis",  # Support for Redis caching in Django
+        "edx-sysadmin",
+        "edx-username-changer==0.2.0",
```
Comment (shaidar) [inline src/bilder/images/edxapp/group_data/all.py:23]: Think the latest is `0.2.1`

File: src/bilder/images/edxapp/group_data/all.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    ],
+    "mitx": [
+        "celery-redbeat",  # Support for using Redis as the lock for Celery schedules
+        "django-redis",  # Support for Redis caching in Django
+        "edx-sga==0.17.2",
+        "edx-sysadmin",
+        "git+https://github.com/raccoongang/xblock-pdf.git@8d63047c53bc8fdd84fa7b0ec577bb0a729c215f#egg=xblock-pdf",
+        "ol-openedx-logging",
+        "ol-openedx-sentry",
+        "rapid-response-xblock==0.1.0",
```
Comment (shaidar) [inline src/bilder/images/edxapp/group_data/all.py:47]: Think this should be `0.6.0`

File: src/bilder/images/edxapp/edxapp_base.pkr.hcl [hunk 6]
```diff
@@ -1,32 +1,32 @@
 locals {
-  timestamp = regex_replace(timestamp(), "[- TZ:]", "")
+  timestamp     = regex_replace(timestamp(), "[- TZ:]", "")
```
Comment (shaidar) [inline src/bilder/images/edxapp/edxapp_base.pkr.hcl:2]: Spacing?

File: src/bilder/images/edxapp/edxapp_base.pkr.hcl [hunk 7]
```diff
@@ -92,7 +92,7 @@ build {
   }
   provisioner "shell" {
     environment_vars = [
-      "EDX_ANSIBLE_BRANCH=${var.ansible_branch}",
+      "EDX_ANSIBLE_BRANCH=${var.edx_platform_version}",
```
Comment (shaidar) [inline src/bilder/images/edxapp/edxapp_base.pkr.hcl:95]: Should this be named `EDX_PLATFORM_VERSION`?

File: src/bilder/images/edxapp/plugins/git_export_import.py [hunk 8]
```diff
@@ -1,28 +1,38 @@
...
-for host in ("github.com", "github.mit.edu"):
-    ssh.keyscan(
-        name=f"Add {host} public SSH fingerprint for course import/export",
-        hostname=host,
-        sudo_user="www-data",
+@deploy("Set up git auto export")
+def git_auto_export(state=None, host=None):
+    pip.packages(
+        name="Install edx-git-auto-export",
+        packages=[
```
Comment (shaidar) [inline src/bilder/images/edxapp/plugins/git_export_import.py:11]: Can't this be part of the other pkgs?

File: src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml [hunk 9]
```diff
@@ -0,0 +1,601 @@
...
+    JWT_SECRET_KEY: {{ .Data.django_secret_key }}
+    JWT_SIGNING_ALGORITHM: RS512
+    JWT_ISSUERS:
+      - ISSUER: https://{{ key "edxapp/lms-domain" }}/oauth2
+        AUDIENCE: mitx
+        SECRET_KEY: {{ .Data.django_secret_key }}
+XQUEUE_INTERFACE:
+    basic_auth:
+    - edx
+    - edx
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml:52]: Duplicate?

File: src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml [hunk 10]
```diff
@@ -0,0 +1,601 @@
...
+    country: hidden
+    gender: optional
+    goals: optional
+    honor_code: required
+    level_of_education: optional
+    mailing_address: hidden
+    terms_of_service: hidden
+    year_of_birth: optional
+RETIRED_EMAIL_DOMAIN: retired.invalid
+RETIRED_EMAIL_PREFIX: retired__user_
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml:458]: Is this right?

File: src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml [hunk 11]
```diff
@@ -0,0 +1,601 @@
...
+- COMPLETE
+SEGMENT_KEY: null
+SENTRY_ENVIRONMENT: {{ env "ENVIRONMENT" }}
+SERVER_EMAIL: odl-devops@mit.edu # MODIFIED
+# Removing the session cookie domain as it is no longer needed for sharing the cookie
+# between LMS and Studio (TMM 2021-10-22)
+# SESSION_COOKIE_DOMAIN: {{ key "edxapp/cookie-domain" }} # MODIFIED
+SESSION_COOKIE_SECURE: true # MODIFIED
+SESSION_SAVE_EVERY_REQUEST: false
+SITE_NAME: MITx Online  # MODIFIED
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml:477]: Isn't this residential?

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 12]
```diff
@@ -0,0 +1,577 @@
...
+        VMEM: 536870912
+    python_bin: /edx/app/edxapp/venvs/edxapp-sandbox/bin/python
+    user: sandbox
+COMMENTS_SERVICE_URL: http://localhost:4567  # MODIFIED
+COMPREHENSIVE_THEME_DIRS:
+- /edx/app/edxapp/themes/  # MODIFIED
+COMPREHENSIVE_THEME_LOCALE_PATHS: []
+CONTACT_EMAIL: {{ key "edxapp/sender-email-address" }} # MODIFIED
+CORS_ORIGIN_ALLOW_ALL: false
+CORS_ORIGIN_WHITELIST:
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:202]: Should `preview` be added?

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 13]
```diff
@@ -0,0 +1,577 @@
...
+    ENABLE_MKTG_SITE: false
+    ENABLE_MOBILE_REST_API: true
+    ENABLE_OAUTH2_PROVIDER: true
+    ENABLE_PUBLISHER: false
+    ENABLE_READING_FROM_MULTIPLE_HISTORY_TABLES: true
+    ENABLE_SPECIAL_EXAMS: true # MODIFIED FOR PROCTORTRACK
+    ENABLE_THIRD_PARTY_AUTH: true
+    ENABLE_UNICODE_USERNAME: true  # ADDED KEY
+    ENABLE_VIDEO_UPLOAD_PIPELINE: true # MODIFIED
+    MILESTONES_APP: true # ADDED KEY FOR PROCTORTRACK
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:283]: Not using PT

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 14]
```diff
@@ -0,0 +1,577 @@
...
+    INDIVIDUAL_DUE_DATES: true
+    LICENSING: true
+    MILESTONES_APP: true
+    REQUIRE_COURSE_EMAIL_AUTH: false
+    REROUTE_ACTIVATION_EMAIL: support@xpro.mit.edu
+    RESTRICT_ENROLL_BY_REG_METHOD: false
+    SESSION_COOKIE_SECURE: true
+FEEDBACK_SUBMISSION_EMAIL: ''
+FERNET_KEYS:
+- DUMMY KEY CHANGE BEFORE GOING TO PRODUCTION
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:315]: Should this have a value stored in vault?

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 15]
```diff
@@ -0,0 +1,577 @@
...
+    ENFORCE_COMPLIANCE_ON_LOGIN: false
+PASSWORD_RESET_SUPPORT_LINK: ''
+PAYMENT_SUPPORT_EMAIL: {{ key "edxapp/sender-email-address" }}  # MODIFIED
+PLATFORM_DESCRIPTION: MIT xPRO # MODIFIED
+PLATFORM_FACEBOOK_ACCOUNT: http://www.facebook.com/YourPlatformFacebookAccount
+PLATFORM_NAME: MIT xPRO  # MODIFIED
+PLATFORM_TWITTER_ACCOUNT: '@YourPlatformTwitterAccount'
+POLICY_CHANGE_GRADES_ROUTING_KEY: edx.lms.core.default
+PRESS_EMAIL: support@xpro.mit.edu # MODIFIED
+PROCTORING_BACKENDS:
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:407]: This is not needed

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 16]
```diff
@@ -0,0 +1,577 @@
...
+    CERTIFICATE_FACEBOOK: false
+    CERTIFICATE_TWITTER: false
+    CUSTOM_COURSE_URLS: false
+    DASHBOARD_FACEBOOK: false
+    DASHBOARD_TWITTER: false
+STATIC_ROOT_BASE: /edx/var/edxapp/staticfiles
+STATIC_URL_BASE: /static/
+STUDIO_NAME: Studio
+STUDIO_SHORT_NAME: Studio
+SUPPORT_SITE_LINK: https://mitx-micromasters.zendesk.com/hc/en-us/requests/new
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:461]: ?

File: src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml [hunk 17]
```diff
@@ -0,0 +1,601 @@
...
+    CERTIFICATE_FACEBOOK: false
+    CERTIFICATE_TWITTER: false
+    CUSTOM_COURSE_URLS: false
+    DASHBOARD_FACEBOOK: false
+    DASHBOARD_TWITTER: false
+STATIC_ROOT_BASE: /edx/var/edxapp/staticfiles
+STATIC_URL_BASE: /static/
+STUDIO_NAME: Studio
+STUDIO_SHORT_NAME: Studio
+SUPPORT_SITE_LINK: https://mitx-micromasters.zendesk.com/hc/en-us/requests/new
```
Comment (shaidar) [inline src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml:489]: ?

File: src/bilder/images/edxapp/deploy.py [hunk 18]
```diff
@@ -113,31 +112,27 @@
...
+
+# Set up Consul templates
+lms_config_path = Path("/edx/etc/lms.yml")
+studio_config_path = Path("/edx/etc/studio.yml")
+forum_config_path = Path("/edx/app/forum/forum_env")
+lms_intermediate_template = Path("/etc/consul-template/templates/edxapp-lms.tmpl")
+studio_intermediate_template = Path("/etc/consul-template/templates/edxapp-studio.tmpl")
+forum_intermediate_template = Path("/etc/consul-template/templates/edx-forum.tmpl")
 consul_templates = [
-    ConsulTemplateTemplate(
```
Comment (shaidar) [inline src/bilder/images/edxapp/deploy.py:117]: Why is this removed?

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 19]
```diff
@@ -0,0 +1,577 @@
...
+    ALLOW_ALL_ADVANCED_COMPONENTS: true
+    ALLOW_COURSE_STAFF_GRADE_DOWNLOADS: true
+    AUTH_USE_CERTIFICATES: false
+    BYPASS_ACTIVATION_EMAIL_FOR_EXTAUTH: true
+    CERTIFICATES_ENABLED: false
+    DISABLE_LOGIN_BUTTON: false
+    ENABLE_AUTO_COURSE_REGISTRATION: true
+    ENABLE_COURSE_BLOCKS_NAVIGATION_API: true
+    ENABLE_INSTRUCTOR_EMAIL: true
+    ENABLE_INSTRUCTOR_LEGACY_DASHBOARD: true
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:299]: I think this has been removed

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 20]
```diff
@@ -0,0 +1,577 @@
...
+    ENABLE_INSTRUCTOR_REMOTE_GRADEBOOK_CONTROLS: true
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:300]: don't need this for xPRO

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 21]
```diff
@@ -0,0 +1,577 @@
...
+    ENABLE_PAID_COURSE_REGISTRATION: false
+    ENABLE_PREREQUISITE_COURSES: true
+    ENABLE_RENDER_XBLOCK_API: true
+    ENABLE_SHOPPING_CART: true
+    ENABLE_SYSADMIN_DASHBOARD: true
+    INDIVIDUAL_DUE_DATES: true
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:306]: don't need this any more.

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 22]
```diff
@@ -0,0 +1,577 @@
...
+    LICENSING: true
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:307]: actually, should prob be false for xPRO, but doesn't matter too much since it also needs to be set per-course.

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml (hunk 14 above)
Comment (pdpinch) [reply src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:315]: What's this for / is it documented anywhere?

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 23]
```diff
@@ -0,0 +1,577 @@
...
+FERNET_KEYS:
+- DUMMY KEY CHANGE BEFORE GOING TO PRODUCTION
+FILE_UPLOAD_STORAGE_BUCKET_NAME: {{ key "edxapp/s3-storage-bucket" }}
+FILE_UPLOAD_STORAGE_PREFIX: submissions_attachments
+FINANCIAL_REPORTS:
+    BUCKET: null
+    ROOT_PATH: sandbox
+    STORAGE_TYPE: localfs
+FOOTER_ORGANIZATION_IMAGE: images/logo.png
+GITHUB_REPO_ROOT: /edx/var/edxapp/data
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:323]: I think this can be removed, but I"ll check.

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 24]
```diff
@@ -0,0 +1,577 @@
...
+    level_of_education: optional
+    mailing_address: hidden
+    terms_of_service: hidden
+    year_of_birth: optional
+RETIRED_EMAIL_DOMAIN: retired.invalid
+RETIRED_EMAIL_PREFIX: retired__user_
+RETIRED_USERNAME_PREFIX: retired__user_
+RETIRED_USER_SALTS:
+- OVERRIDE ME WITH A RANDOM VALUE
+- ROTATE SALTS BY APPENDING NEW VALUES
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:434]: should we set this?

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml (hunk 16 above)
Comment (pdpinch) [reply src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:461]: should be https://xpro.zendesk.com/hc I believe

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 25]
```diff
@@ -0,0 +1,577 @@
...
+    VIDEO_TRANSCRIPTS_MAX_BYTES: 3145728
+VIDEO_UPLOAD_PIPELINE:
+    BUCKET: ''
+    ROOT_PATH: ''
+WIKI_ENABLED: true
+XBLOCK_FS_STORAGE_BUCKET: null
+XBLOCK_FS_STORAGE_PREFIX: null
+XBLOCK_SETTINGS: {}
+X_FRAME_OPTIONS: DENY
+YOUTUBE_API_KEY: PUT_YOUR_API_KEY_HERE
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:492]: We get frequent errors from YouTube about API access. Maybe they'll go way if we omit this, or leave it as null or empty?

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 26]
```diff
@@ -0,0 +1,577 @@
...
+  expiration_datetime: null
+  min_price: 0
+  sku: null
+  slug: no-id-professional
+  suggested_prices:
+EMAIL_HOST: smtp.mailgun.org
+EMAIL_PORT: 587
+EMAIL_USE_TLS: true
+FIELD_OVERRIDE_PROVIDERS:
+  - lms.djangoapps.courseware.student_field_overrides.IndividualStudentOverrideProvider
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:577]: I'm pretty sure this is no longer needed.

File: src/bilder/images/edxapp/templates/edxapp/xpro/lms_only.yml [hunk 27]
```diff
@@ -0,0 +1,124 @@
...
+- bulk_email_optout
+ACE_ROUTING_KEY: edx.lms.core.default
+# ALTERNATE_WORKER_QUEUES: cms  # The code already has this default
+# ANALYTICS_API_KEY: ''
+# ANALYTICS_API_URL: http://localhost:18100
+API_ACCESS_FROM_EMAIL: api-requests@example.com
+API_ACCESS_MANAGER_EMAIL: api-access@example.com
+API_DOCUMENTATION_URL: http://course-catalog-api-guide.readthedocs.io/en/latest/
+AUDIT_CERT_CUTOFF_DATE: null
+AUTH_DOCUMENTATION_URL: http://course-catalog-api-guide.readthedocs.io/en/latest/authentication/index.html
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/lms_only.yml:21]: I don't think we need these

File: src/bilder/images/edxapp/templates/edxapp/xpro/lms_only.yml [hunk 28]
```diff
@@ -0,0 +1,124 @@
...
+BULK_EMAIL_ROUTING_KEY_SMALL_JOBS: edx.lms.core.default
+CONTACT_MAILING_ADDRESS: SET-ME-PLEASE
+CREDIT_HELP_LINK_URL: ''
+DCS_SESSION_COOKIE_SAMESITE: Lax
+DCS_SESSION_COOKIE_SAMESITE_FORCE_ALL: true
+EDXNOTES_INTERNAL_API: http://localhost:18120/api/v1
+EDXNOTES_PUBLIC_API: http://localhost:18120/api/v1
+# EDX_API_KEY: PUT_YOUR_API_KEY_HERE  # DEPRECATED SETTING
+GIT_REPO_DIR: /edx/var/edxapp/course_repos
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/lms_only.yml:30]: not sure if we need this. I'll check

File: src/bilder/images/edxapp/templates/edxapp/xpro/lms_only.yml [hunk 29]
```diff
@@ -0,0 +1,124 @@
...
+    '
+PDF_RECEIPT_FOOTER_TEXT: 'Enter your receipt footer text here.
+
+    '
+PDF_RECEIPT_LOGO_PATH: ''
+PDF_RECEIPT_TAX_ID: 00-0000000
+PDF_RECEIPT_TAX_ID_LABEL: fake Tax ID
+PDF_RECEIPT_TERMS_AND_CONDITIONS: 'Enter your receipt terms and conditions here.
+
+    '
```
Comment (pdpinch) [inline src/bilder/images/edxapp/templates/edxapp/xpro/lms_only.yml:67]: we're not making receipts, so shouldn't need any of this

**Review by pdpinch — CHANGES_REQUESTED** [review]
More questions than answers. The only thing I think you must change is `SUPPORT_SITE_LINK` for xPRO

File: src/bilder/images/edxapp/deploy.py (hunk 18 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/deploy.py:117]: The commit message describes the situation, but the idea is that by splitting the Packer builds into the Ansible run and the customization it allows us to build new instances for config changes more quickly, so that we don't have to use the 2 step consul-template rendering that we have been doing.

File: src/bilder/images/edxapp/packer_vars/mitx.pkrvars.hcl (hunk 3 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/packer_vars/mitx.pkrvars.hcl:1]: No, because the AMI only gets built once and used across environments. We're just using the CI build environment so that it doesn't run any risk of interfering with other environments.

File: src/bilder/images/edxapp/plugins/git_export_import.py (hunk 8 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/plugins/git_export_import.py:11]: It _can_, but I set this up as its own operation because of the other steps that are necessary to make it work.

File: src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml (hunk 9 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml:52]: No, just a really bad placeholder value...

File: src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml [hunk 30]
```diff
@@ -0,0 +1,601 @@
...
+
+ACTIVATION_EMAIL_SUPPORT_LINK: ''
+AFFILIATE_COOKIE_NAME: dev_affiliate_id
+# ANALYTICS_DASHBOARD_NAME: Your Platform Name Here Insights
+# ANALYTICS_DASHBOARD_URL: http://localhost:18110/courses
+AUTH_PASSWORD_VALIDATORS:
+-   NAME: django.contrib.auth.password_validation.UserAttributeSimilarityValidator
+-   NAME: common.djangoapps.util.password_policy_validators.MinimumLengthValidator
+    OPTIONS:
+        min_length: 2
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml:123]: Probably. This is just the default values that are spit out by Ansible and which we've never modified.

File: src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml (hunk 10 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/mitx/common_values.yml:458]: Another default value from Ansible. I think this is only used for GDPR requests

File: src/bilder/images/edxapp/templates/edxapp/mitx/lms_only.yml [hunk 31]
```diff
@@ -0,0 +1,118 @@
...
+- bulk_email_optout
+ACE_ROUTING_KEY: edx.lms.core.default
+# ALTERNATE_WORKER_QUEUES: cms  # The code already has this default
+# ANALYTICS_API_KEY: ''
+# ANALYTICS_API_URL: http://localhost:18100
+API_ACCESS_FROM_EMAIL: api-requests@example.com
+API_ACCESS_MANAGER_EMAIL: api-access@example.com
+API_DOCUMENTATION_URL: http://course-catalog-api-guide.readthedocs.io/en/latest/
+AUDIT_CERT_CUTOFF_DATE: null
+AUTH_DOCUMENTATION_URL: http://course-catalog-api-guide.readthedocs.io/en/latest/authentication/index.html
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/mitx/lms_only.yml:16]: Probably not, these are just default values that were generated from the Ansible run that I used to baseline the configs and haven't revisited it since then.

File: src/bilder/images/edxapp/templates/edxapp/mitx/lms_only.yml [hunk 32]
```diff
@@ -0,0 +1,118 @@
...
+PDF_RECEIPT_DISCLAIMER_TEXT: 'ENTER YOUR RECEIPT DISCLAIMER TEXT HERE.
+
+    '
+PDF_RECEIPT_FOOTER_TEXT: 'Enter your receipt footer text here.
+
+    '
+PDF_RECEIPT_LOGO_PATH: ''
+PDF_RECEIPT_TAX_ID: 00-0000000
+PDF_RECEIPT_TAX_ID_LABEL: fake Tax ID
+PDF_RECEIPT_TERMS_AND_CONDITIONS: 'Enter your receipt terms and conditions here.
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/mitx/lms_only.yml:60]: More default values that I haven't taken the time to try and get rid of.

File: src/bilder/images/edxapp/templates/edxapp/mitx/lms_only.yml [hunk 33]
```diff
@@ -0,0 +1,118 @@
...
+RATELIMIT_RATE: '600/m'  # ADDED KEY
+RECALCULATE_GRADES_ROUTING_KEY: edx.lms.core.default
+SESSION_COOKIE_NAME: {{ env "ENVIRONMENT" }}-edx-lms-sessionid  # MODIFIED
+THIRD_PARTY_AUTH_BACKENDS:
+- social_core.backends.google.GoogleOAuth2
+- social_core.backends.linkedin.LinkedinOAuth2
+- social_core.backends.facebook.FacebookOAuth2
+- social_core.backends.azuread.AzureADOAuth2
+- common.djangoapps.third_party_auth.appleid.AppleIdAuth
+- common.djangoapps.third_party_auth.identityserver3.IdentityServer3
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/mitx/lms_only.yml:89]: No, just didn't take it out from the default list. For residential we should be able to just have SAML. I'll tweak that as we go along.

File: src/bilder/images/edxapp/templates/edxapp/mitxonline/common_values.yml [hunk 34]
```diff
@@ -284,7 +284,6 @@ FEATURES:
     SHOW_HEADER_LANGUAGE_SELECTOR: false
     SKIP_EMAIL_VALIDATION: true # ADDED KEY
     ALLOW_PUBLIC_ACCOUNT_CREATION: true
-    ALLOW_COURSE_STAFF_GRADE_DOWNLOADS: true # ADDED KEY
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/mitxonline/common_values.yml:287]: That's an accidental deletion. I'll add it back in.

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml (hunk 12 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:195]: Yes, will fix that now.

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 35]
```diff
@@ -0,0 +1,577 @@
...
+  - https://{{ key "edxapp/lms-domain" }}
+  - https://{{ key "edxapp/studio-domain" }}
+  - https://{{ key "edxapp/marketing-domain" }}
+COURSES_WITH_UNSAFE_CODE: []
+COURSE_ABOUT_VISIBILITY_PERMISSION: see_exists
+COURSE_CATALOG_API_URL: http://localhost:8008/api/v1
+COURSE_CATALOG_URL_ROOT: http://localhost:8008
+COURSE_CATALOG_VISIBILITY_PERMISSION: see_exists
+CREDENTIALS_INTERNAL_SERVICE_URL: http://localhost:8005
+CREDENTIALS_PUBLIC_SERVICE_URL: http://localhost:8005
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:205]: No, this is another default setting that I haven't bothered to remove.

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml [hunk 36]
```diff
@@ -0,0 +1,577 @@
...
+# EMAIL_HOST_PASSWORD: ''
+# EMAIL_HOST_USER: ''
+# EMAIL_PORT: 25
+# EMAIL_USE_TLS: false
+ENABLE_COMPREHENSIVE_THEMING: true  # MODIFIED
+EVENT_TRACKING_SEGMENTIO_EMIT_WHITELIST: []
+EXTRA_MIDDLEWARE_CLASSES: []
+FACEBOOK_API_VERSION: v2.1
+FACEBOOK_APP_ID: FACEBOOK_APP_ID
+FACEBOOK_APP_SECRET: FACEBOOK_APP_SECRET
```
[the remark this answers is not in the document]
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:243]: Most likely not, more defaults. Would be helpful at some point to go through and clear out even more unnecessary settings and update edx-platform to have better defaults.

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml (hunk 24 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:425]: I'm not sure. I think this is for use by the GDPR user deletion feature. That will require some investigation.

File: src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml (hunk 25 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/templates/edxapp/xpro/common_values.yml:483]: I'll set it as null

File: src/bilder/components/hashicorp/templates/consul-template.service.j2 [hunk 37]
```diff
@@ -1,7 +1,7 @@
 [Unit]
 Description=consul agent
 Requires=network-online.target
-After=network-online.target
+After=network-online.target vault.service consul.service
```
Comment (shaidar) [inline src/bilder/components/hashicorp/templates/consul-template.service.j2:4]: Is the `vault.service` for the vault agent running on the consul cluster instances?

File: src/bilder/images/consul/deploy.py [hunk 38]
```diff
@@ -41,6 +41,7 @@
     Path("00-default.json"): ConsulConfig(
         bootstrap_expect=3,
         server=True,
+        ui=True,
```
Comment (shaidar) [inline src/bilder/images/consul/deploy.py:44]: ?

File: src/bilder/images/edxapp/packer_vars/xpro.pkrvars.hcl [hunk 39]
```diff
@@ -0,0 +1,4 @@
+build_environment    = "mitxpro-ci"
+edx_platform_version = "open-release/maple.master"
+installation_target  = "xpro"
```
Comment (shaidar) [inline src/bilder/images/edxapp/packer_vars/xpro.pkrvars.hcl:3]: Should this be `mitxpro`?

File: src/bilder/components/hashicorp/templates/consul-template.service.j2 (hunk 37 above)
Comment (blarghmatey) [reply src/bilder/components/hashicorp/templates/consul-template.service.j2:4]: Correct

File: src/bilder/images/edxapp/packer_vars/xpro.pkrvars.hcl (hunk 39 above)
Comment (blarghmatey) [reply src/bilder/images/edxapp/packer_vars/xpro.pkrvars.hcl:4]: No, we are using xpro for the naming of the environment but mitxpro for the OU tags.

File: src/bilder/images/consul/deploy.py (hunk 38 above)
Comment (blarghmatey) [reply src/bilder/images/consul/deploy.py:44]: That's so that we can use the UI for the Consul servers to identify registered services since we'll be moving away from having the Salt master running for being able to query state that way.
